lib/lib.go

```diff
@@ -0,0 +1,189 @@
+type Podium struct {
+	Config            *viper.Viper
+	URL               string
+	baseLeaderboard   string
+	localeLeaderboard string
+}
+
+// NewPodium returns a new podium API application
+func NewPodium(config *viper.Viper) *Podium {
+	app := &Podium{
+		Config:            config,
+		URL:               config.GetString("podium.url"),
+		baseLeaderboard:   config.GetString("leaderboards.globalLeaderboard"),
+		localeLeaderboard: config.GetString("leaderboards.localeLeaderboard"),
+	}
+	return app
+}
+
+func sendTo(method, url string, payload map[string]interface{}) (int, string, error) {
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		return -1, "", err
+	}
+
+	var req *http.Request
+
+	if payload != nil {
+		req, err = http.NewRequest(method, url, bytes.NewBuffer(payloadJSON))
+		if err != nil {
+			return -1, "", err
+		}
+	} else {
+		req, err = http.NewRequest(method, url, nil)
+		if err != nil {
+			return -1, "", err
+		}
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	defer resp.Body.Close()
+	if err != nil {
+		return -1, "", err
+	}
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	return resp.StatusCode, string(body), nil
+}
+
+func (app *Podium) buildURL(pathname string) string {
+	return fmt.Sprintf("%s%s", app.URL, pathname)
+}
+
+func (app *Podium) buildDeleteLeaderboardURL(leaderboard string) string {
+	var pathname = fmt.Sprintf("/l/%s", leaderboard)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildGetTopPercentURL(leaderboard string, percentage int) string {
+	var pathname = fmt.Sprintf("/l/%s/top-percent/%d", leaderboard, percentage)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildUpdateScoreURL(leaderboard string, playerID string) string {
+	var pathname = fmt.Sprintf("/l/%s/members/%s/score", leaderboard, playerID)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildIncrementScoreURL(leaderboard string, playerID string) string {
+	return app.buildUpdateScoreURL(leaderboard, playerID)
+}
+
+func (app *Podium) buildUpdateScoresURL(playerID string) string {
+	var pathname = fmt.Sprintf("/m/%s/scores", playerID)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildRemoveMemberFromLeaderboardURL(leaderboard string, member string) string {
+	var pathname = fmt.Sprintf("/l/%s/members/%s", leaderboard, member)
+	return app.buildURL(pathname)
+}
+
+// page is 1-based
+func (app *Podium) buildGetTopURL(leaderboard string, page int, pageSize int) string {
+	var pathname = fmt.Sprintf("/l/%s/top/%d?pageSize=%d", leaderboard, page, pageSize)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildGetPlayerURL(leaderboard string, playerID string) string {
+	var pathname = fmt.Sprintf("/l/%s/members/%s", leaderboard, playerID)
+	return app.buildURL(pathname)
+}
+
+func (app *Podium) buildHealthcheckURL() string {
+	var pathname = "/healthcheck"
+	return app.buildURL(pathname)
+}
+
+// external functions:
```

Comment:
In Go, all public methods should be commented, see: https://blog.golang.org/godoc-documenting-go-code

You should install a go linter in your text editor, it would've warned you about that.